secboot/secboot_hooks.go

```diff
@@ -185,6 +185,15 @@ func unlockVolumeUsingSealedKeyFDERevealKeyV2(name, sealedEncryptionKeyFile, sou
 	if err != nil {
 		return res, fmt.Errorf("cannot unlock encrypted partition: %v", err)
 	}
+	// ensure we close the open volume under any error condition
+	defer func() {
+		if err != nil {
+			if err := sbDeactivateVolume(mapperName); err != nil {
+				logger.Noticef("cannot deactivate volume %q", mapperName)
```

Comment:
should this also include the error we get from secboot about deactivating the volume?

```suggestion
				logger.Noticef("cannot deactivate volume %q: %v", mapperName, err)
```

Reply:
Thinking about this I guess if we are paranoid we could panic() here if we can't deactivate the volume to avoid that anyone gets the chance to trick the system into continuing to boot. But maybe that is too extreme?